tsdb/engine/tsm1/compact.go

```diff
@@ -610,9 +641,16 @@ func (c *DefaultPlanner) Plan(lastWrite time.Time) ([]CompactionGroup, int64) {
 		// can become larger faster than ones after them.  We want to skip those really big ones and just
 		// compact the smaller ones until they are closer in size.
 		if i > 0 {
-			if g.size()*2 < generations[i-1].size() {
-				start = i
-				break
+			if c.enableNestedCompactor {
+				if g.size()*2 < generations[i-1].size() && generations[i-1].level() >= generations[i].level() {
```

Comment:
Going to add a comment about why this check needs to be different with the feature flag on. It's a bit confusing to just look at without knowing context.